### PR TITLE
feat(MySQLdb): add tracing for `connect`

### PR DESCRIPTION
**Pull request by p7g**
While evaluating the performance of a Django app, I wanted to see how much time was spent creating new database connections, which required patching MySQLdb myself. It seemed simple enough to add that tracing to the existing MySQLdb patches, so here we are.

To test it I installed my local dd-trace-py with these changes and made sure the span showed up.

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/contrib/mysqldb/patch.py [hunk 1]
```diff
@@ -58,7 +59,8 @@ def unpatch():
 
 
 def _connect(func, instance, args, kwargs):
-    conn = func(*args, **kwargs)
+    with tracer.trace("MySQLdb.connection.connect", service=config.mysqldb._default_service):
```
Comment (brettlangdon) [inline ddtrace/contrib/mysqldb/patch.py:62]: can we add a `trace_connect` option to the config above on L22.

to keep consistent with the existing behavior.

**Review by brettlangdon — COMMENTED** [review]
Thanks for the PR, can you add a release note and some tests?

File: tests/contrib/mysqldb/test_mysqldb.py [hunk 2]
```diff
@@ -509,16 +509,19 @@ def _connect_with_kwargs(self):
             }
         )
 
+    def _add_dummy_tracer_to_pinned(self, obj):
```
Comment (Kyle-Verhoog) [inline tests/contrib/mysqldb/test_mysqldb.py:512]: This is probably better suited for `setUp` and then undo it in `tearDown`

File: ddtrace/contrib/mysqldb/patch.py [hunk 3]
```diff
@@ -19,6 +22,7 @@
         _default_service="mysql",
         _dbapi_span_name_prefix="mysql",
         trace_fetch_methods=asbool(os.getenv("DD_MYSQLDB_TRACE_FETCH_METHODS", default=False)),
+        trace_connect=asbool(os.getenv("DD_MYSQLDB_TRACE_CONNECT", default=False)),
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/mysqldb/patch.py:25]: we should also document this option in the integration docs

File: tests/contrib/mysqldb/test_mysqldb.py [hunk 4]
```diff
@@ -614,3 +617,24 @@ def test_user_specified_service_integration(self):
...
+
+    def test_trace_connect(self):
+        self._add_dummy_tracer_to_pinned(MySQLdb)
+
+        # No span when trace_connect is False (the default)
+        self._connect_with_kwargs().close()
+        spans = self.tracer.pop()
+        assert not spans
+
+        with self.override_config("mysqldb", dict(trace_connect=True)):
```
Comment (Kyle-Verhoog) [inline tests/contrib/mysqldb/test_mysqldb.py:629]: Worth testing the environment variable too (see the `test_user_specified_service_integration` above for a good way to do this)

**Review by Kyle-Verhoog — COMMENTED** [review]
Also just needs a release note, otherwise lgtm!

**Review by Kyle-Verhoog — APPROVED** [review]

File: releasenotes/notes/mysqldb-trace-connect-53686ab262bf37d5.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    MySQLdb: Added optional tracing for MySQLdb.connect, using the configuration option
```
Comment (brettlangdon) [inline releasenotes/notes/mysqldb-trace-connect-53686ab262bf37d5.yaml:4]:
```suggestion
    MySQLdb: Added optional tracing for ``MySQLdb.connect``, using the configuration option
```

**Review by brettlangdon — APPROVED** [review]
non-blocking nit for the release note

**Pull request merged**